### PR TITLE
clean tmater client for setting instance info

### PR DESCRIPTION
**Pull request by nlu90**
Based on the discussion in: #2291, this PR is trying to clean the logic in tmater client

### CODE REVIEW

**Comment by srkukarni** [comment]
I believe the rebase was done incorrectly. It should be instance_server and not server_

File: heron/stmgr/src/cpp/manager/tmaster-client.cpp [hunk 1]
```diff
@@ -210,31 +214,42 @@ void TMasterClient::OnHeartbeatTimer() {
...
-    stmgr->set_data_port(data_port_);
-    stmgr->set_local_data_port(local_data_port_);
-    stmgr->set_local_endpoint("/unused");
-    stmgr->set_cwd(cwd);
-    stmgr->set_pid((sp_int32)ProcessUtils::getPid());
-    stmgr->set_shell_port(shell_port_);
+void TMasterClient::SetInstanceInfo(const std::vector<proto::system::Instance*>& _instances) {
     for (auto iter = _instances.begin(); iter != _instances.end(); ++iter) {
-      register_request_.add_instances()->CopyFrom(*(*iter));
+      auto instance = new proto::system::Instance();
```
Comment (srkukarni) [inline heron/stmgr/src/cpp/manager/tmaster-client.cpp]: 1. We need to Clean instances_ if it already exists
2. for (auto iter = _instances.begin(); iter != _instances.end(); ++iter) {
    auto instance = new proto::system::Instance();
    instance->CopyFrom(*iter);
    instances_.push_back(instance)
   }

File: heron/stmgr/src/cpp/manager/tmaster-client.cpp [hunk 2]
```diff
@@ -210,31 +214,42 @@ void TMasterClient::OnHeartbeatTimer() {
 }
 
 void TMasterClient::SendRegisterRequest() {
-  CHECK(register_request_set_);
-  proto::tmaster::StMgrRegisterRequest* request =
-    new proto::tmaster::StMgrRegisterRequest(register_request_);
+  auto request = new proto::tmaster::StMgrRegisterRequest();
```
Comment (srkukarni) [inline heron/stmgr/src/cpp/manager/tmaster-client.cpp]: Please rebase with master

Comment (nlu90) [reply heron/stmgr/src/cpp/manager/tmaster-client.cpp]: rebased.

**Comment by nlu90** [comment]
@srkukarni The rebase work causes confusions to the PR. It is closed accidentally when i tried to update the branch. I'll create a new PR for this work and reference this pr.

**Pull request closed without merging**